aiohttp/http_parser.py

```diff
@@ -729,13 +729,13 @@ def feed_data(
 
             if required >= chunk_len:
                 self._length = required - chunk_len
-                self.payload.feed_data(chunk, chunk_len)
+                self.payload.feed_data(chunk)
                 if self._length == 0:
                     self.payload.feed_eof()
                     return True, b""
             else:
                 self._length = 0
-                self.payload.feed_data(chunk[:required], required)
+                self.payload.feed_data(chunk[:required])
                 self.payload.feed_eof()
                 return True, chunk[required:]
```

Comment:
Maybe some further improvement to do this:
```
self._length = max(required - len(chunk), 0)
self.payload.feed_data(chunk[:required])
if self._length == 0:
    self.payload.feed_eof()
    return True, chunk[required:]
```

And remove the chunk_len variable on the line above.